src/components/SettingsModal.js

```diff
@@ -0,0 +1,159 @@
+            <Controller
+              name={componentProps.name.minAmountForFreeDelivery}
+              control={control}
+              rules={{ required: 'Valor minimo em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Valor minimo para entrega grátis'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.minAmountForFreeDelivery ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.deliveryFee}
+              control={control}
+              rules={{ required: 'Valor de taxa entrega em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Valor de taxa entrega'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.deliveryFee ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.discount}
+              control={control}
+              rules={{ required: 'Desconto em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Percentagem de desconto'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.discount ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              control={control}
+              name={componentProps.name.haveGlobalDeliveryDiscount}
+              render={({ field }) => <FormControlLabel control={<Switch checked={haveGlobalDeliveryDiscount} {...field} />} label='Entrega gratuita geral' />}
+            />
+
+            <button type={componentProps.type.submit} ref={submitFormRef} hidden>
+              Criar
+            </button>
+          </form>
+        </Box>
+        {errorMessage && <ErrorMessage message={errorMessage} />}
+        {errors.minAmountForFreeDelivery && <ErrorMessage message={errors.minAmountForFreeDelivery.message} />}
+        {errors.haveGlobalDeliveryDiscount && <ErrorMessage message={errors.haveGlobalDeliveryDiscount.message} />}
+        {errors.deliveryFee && <ErrorMessage message={errors.deliveryFee.message} />}
+        {errors.discount && <ErrorMessage message={errors.discount.message} />}
```

Comment:
You can set a error message using react hook query instead of this. You can create a component with label and error and set a some the rules like "Required field". Then if needed you can show a default error message

Take a look here:
https://react-hook-form.com/get-started/#Handleerrors